---
title: C28105 | Microsoft Docs
ms.custom: ''
ms.date: 11/04/2016
ms.technology:
- vs-ide-code-analysis
ms.topic: conceptual
f1_keywords:
- C28105
helpviewer_keywords:
- C28105
ms.assetid: 5afdec78-30eb-470c-980f-93d0cb3e74cc
author: mikeblome
ms.author: mblome
manager: douge
ms.workload:
- multiple
ms.openlocfilehash: f13afe72d53f8765cc7d1e1e2540b23f2d378d56
ms.sourcegitcommit: 6a9d5bd75e50947659fd6c837111a6a547884e2a
ms.translationtype: MT
ms.contentlocale: ko-KR
ms.lasthandoff: 04/16/2018
---
# <a name="c28105"></a>C28105
경고 C28105: 누수 되는 예외로 인해 리소스  
  
 예외가 발생 하는 경우에 지정 된 리소스가 해제 되지 않습니다. 경로 의해 지정 된 문이 예외를 발생 시킬 수 있습니다. 이 경고는 경고 비슷합니다 [C28103](../code-quality/c28103.md)제외 하 고이 경우 예외가 과정이 진행 중입니다.  
  
## <a name="example"></a>예제  
 다음 코드 예제에서는 이 경고를 생성합니다.  
  
```  
res = KeSaveFloatingPointState(buffer);  
  
res = AllocateResource(Resource);  
char *p2 = new char[10]; // could throw  
  
delete[] p2;  
FreeResource(Resource)  
```  
  
 다음 코드 예제에서는 이 경고를 방지합니다.  
  
```  
res = AllocateResource(Resource);  
char *p2;  
  
try {  
    p2 = new char[10];  
} catch (std::bad_alloc *e) {  
    // just handle the throw  
    ;  
}  
FreeResource(Resource)  
```